File: src/Budget.Domain/Entities/Transaction.cs
namespace Budget.Domain.Entities;

public class Transaction
{
    public Guid Id { get; set; }

    public string Description { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public DateTime Date { get; set; }


    // Relationship

    public Guid AccountId { get; set; }

    public Account Account { get; set; } = null!;


    public Guid CategoryId { get; set; }

    public Category Category { get; set; } = null!;
}

File: frontend/src/types/transaction.ts
export type Transaction = {
    id: string;
    description: string;
    amount: number;
    date: string;
    accountName: string;
    categoryName: string;
};

export type CreateTransaction = {
    description: string;
    amount: number;
    date: string;
    accountId: string;
    categoryId: string;
};

export type UpdateTransaction = CreateTransaction;